AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java

```diff
@@ -1369,35 +1391,41 @@ private void filterByTag(List<String> selectedTags, int option) {
     }
 
 
-    private abstract class ListenerWithProgressBar extends TaskListener {
+    private static abstract class ListenerWithProgressBar extends TaskListenerWithContext<CardBrowser>{
+        public ListenerWithProgressBar(CardBrowser browser) {
+            super(browser);
+        }
+
         @Override
-        public void onPreExecute() {
-            showProgressBar();
+        public void actualOnPreExecute(@NonNull CardBrowser browser) {
+            browser.showProgressBar();
         }
     }
 
-    private abstract class ListenerWithProgressBarCloseOnFalse extends ListenerWithProgressBar {
-        private String timber = null;
-
-        public ListenerWithProgressBarCloseOnFalse(String timber) {
-            this.timber = timber;
+    /** Does not leak Card Browser. */
+    private static abstract class ListenerWithProgressBarCloseOnFalse extends ListenerWithProgressBar {
+        private final String mTimber;
+        public ListenerWithProgressBarCloseOnFalse(String timber, CardBrowser browser) {
+            super(browser);
+            mTimber = timber;
         }
 
-        public ListenerWithProgressBarCloseOnFalse() {
+        public ListenerWithProgressBarCloseOnFalse(CardBrowser browser) {
+            this(null, browser);
 		}
 
-        public void onPostExecute(TaskData result) {
-            if (timber != null) {
-                Timber.d(timber);
+        public void actualOnPostExecute(@NonNull CardBrowser browser, TaskData result) {
+            if (mTimber != null) {
+                Timber.d(mTimber);
             }
             if (result.getBoolean()) {
-                actualPostExecute(result);
+                actualActualOnPostExecute(browser, result);
```

Comment:
Maybe? Sounds better than actual actual

```suggestion
                actualOnValidPostExecute(browser, result);
```

Reply:
Fine by me. Not an easy task here